### ISSUE TITLE
Attack by malicious endpoint

### ISSUE THREAD

**Issue by huitema** [martinthomson/train-protocol#11]
We are worried about attacks by the network, but it is also possible for an endpoint to send the 6bits in the header to an arbitrary value. Maybe this is just a variant of the spoofing attack, but it is simpler to implement. And endpoint that wants to sabotage a local network could set the 6bits to indicate "tons of bandwidth" and DOS its own connection...

**Comment by martinthomson** [comment]
Given that the endpoint that can spoil the message is also the endpoint responsible for sending, I'm struggling to see how this would be a problem.

**Comment by huitema** [comment]
Correct. The malicious endpointcannot really do anything useful this way. OTOH, if the endpoint could report a fake value to the peer, tricking that peer into sending too much data.

This looks like a variant of spoofed ACKs.

**Comment by martinthomson** [comment]
Yes, if we provide a feedback signaling mechanism, that will be vulnerable to all sorts of misrepresentation.  Again, we should acknowledge and move on.

**Comment by huitema** [comment]
Not too sure above "move on". We did include in QUIC a defense against spoofed ACK. We could easily include a defense about spoofed reports. For example, in some TRAIN packets, the server puts a specific value in the 6 bits. If the client reports a value greater than that, the server knows that TRAIN cannot be trusted on that path.

**Comment by kazuho** [comment]
Is there a reason to talk specifically about attack from endpoints?

My understanding is that they are indistinguishable from attacks by middleboxes, as the bandwidth information is not covered by AEAD.

It might make sense to point out that the receiver of TRAIN can mount the same attack as middleboxes might, but that would be it, I think.

**Comment by huitema** [comment]
Maybe close this one, but retain the idea that endpoints could try to find out fakers (middlepoints or endpoint) by sending a probe with some arbitrary value, and verify that the echoed value is lower or equal. Kind of like setting holes in the sequence number to catch optimistic ACK.
